waterlog/src/actions/SegmentLeaksByIdActions.js

```diff
@@ -1,27 +1,25 @@
-import {
-	FETCH_SEGMENTS_LEAK
-} from './Types';
+import { FETCH_SEGMENTS_LEAK } from './Types';
 import { Globals } from './../Globals';
 
-export const fetchSegmentsLeaksById = (id) => (dispatch) => {
-	fetch(`${Globals.API_URL}/api/segmentleaks/${id}`) //Change to use either localhost/server
-		.then((res) => res.json())
-		.then((leak) => {
-			fetch(`${Globals.API_URL}/api/segmentleaks/costs/${id}`)
-				.then((res) => res.json())
-				.then((data) => {
-					fetch(`${Globals.API_URL}/api/segmentleaks/litres/${id}`)
-						.then((res) => res.json())
-						.then((usage) => {
-							dispatch({
-								type: FETCH_SEGMENTS_LEAK,
-								payload: {
-									leak,
-									data,
-									usage
-								}
-							});
-						});
-				});
-		});
-};
\ No newline at end of file
+export const fetchSegmentsLeaksById = id => dispatch => {
+  fetch(`${Globals.API_URL}/api/segmentleaks/${id}`) //Change to use either localhost/server
```

Comment:
Just a note: 
What is happening here is called "Promise Hell" (You can google it 😉 but it's when you have several nested promises ,each dependant on the previous one resolving before executing the next, in this case the fetches ).

Technically it is not wrong, but there is better ways of writing this code to make it more readable. Here is an article that I find explains it quite nicely if you are interested https://medium.com/@pyrolistical/how-to-get-out-of-promise-hell-8c20e0ab0513 

(This is not a required change by me, just something you can have a look at to improve your knowledge and code quality)

Reply:
Thank you I will look at it.